flexeval/core/evaluate_chat_response.py

```diff
@@ -199,37 +195,43 @@ def evaluate_chat_response(  # noqa: C901, PLR0912
     for metric in metrics:
         metric_result = metric.evaluate(
             lm_outputs=[output["lm_output"] for output in outputs],
-            references_list=[output["references"] for output in outputs],
-            extra_info_list=[output["extra_info"] for output in outputs],
+            references_list=[output["chat_instance"].references for output in outputs],
+            extra_info_list=[
+                output["chat_instance"].extra_info
+                | {"messages": output["chat_instance"].messages, "tools": output["chat_instance"].tools}
+                for output in outputs
+            ],
         )
         metric.cleanup_resources()
 
         metrics_summary_dict.update(metric_result.summary)
 
         if metric_result.instance_details:
-            for instance_idx, instance_details in enumerate(
-                metric_result.instance_details,
-            ):
+            for instance_idx, instance_details in enumerate(metric_result.instance_details):
                 instance_metrics_list[instance_idx].update(instance_details)
     for instance_metrics, output in zip(instance_metrics_list, outputs):
-        output.update(instance_metrics)
-
-    # Calculate the finish_reason and validation statistics
-    finish_reason_counter = Counter()
-    tool_call_validation_result_counter = Counter()
-    for output in outputs:
-        messages = output["extra_info"]["messages"]
-        for mes in messages:
-            if "finish_reason" in mes:
-                finish_reason_counter[mes["finish_reason"]] += 1
-            if "tool_call_validation_result" in mes:
-                tool_call_validation_result_counter[mes["tool_call_validation_result"]] += 1
-    for finish_reason, count in finish_reason_counter.items():
-        metrics_summary_dict[f"finish_reason_ratio-{finish_reason}"] = count / sum(finish_reason_counter.values())
-    for validation_result, count in tool_call_validation_result_counter.items():
-        metrics_summary_dict[f"tool_call_validation_result_ratio-{validation_result}"] = count / sum(
-            tool_call_validation_result_counter.values()
-        )
+        output["metrics"] = instance_metrics
 
     logger.info(metrics_summary_dict)
-    return metrics_summary_dict, outputs
+
+    # Restructure the output for backward compatibility
+    # Maybe switch to the new structure sometime
+    # new: `{"lm_output": LMOutput, "chat_instance": ChatInstance, "messages": [...], "metrics": {...}}`
+    # current: `{"lm_output": "...", "finish_reason": "...", "task_inputs": {...}, "references": [...], **metrics}`
```

Comment:
To me, the term `new` and `current` look somewhat similar. How about `current_format` and `legacy_format` ?

Reply:
OK, I've changed them to `new` and `legacy`.
The first one is not the output format that is actually used, so I think `new` would be more understandable.

Reply:
Thanks! 

> The first one is not the output format that is actually used

I actually missed this part, sorry.